### PR TITLE
Feat/clippy 57 

### PR DESCRIPTION
**Pull request by patr1ck001**
Once merged, this pull request will allow to remove the Upload button and add loading animation

### CODE REVIEW

**Review by mihaelKampic — COMMENTED** [review]
Please fix lint issues @patr1ck001

File: src/components/FileUpload.js [hunk 1]
```diff
@@ -13,56 +13,39 @@ import toast, { Toaster } from "react-hot-toast";
...
   }, []);
 
   useEffect(() => {
-    if (tryUpload && pdfFile == null) {
-      toast.error("No file have been uploaded. Please upload a file");
-    }
-
     if (pdfFile !== null && pdfFile.type !== "application/pdf") {
-      toast.error("please upload a pdf file");
+      toast.error("please upload a pdf file", {
```
Comment (mihaelKampic) [inline src/components/FileUpload.js:23]: What will happen if the file upload fails? Do we show a toast message then?

File: src/components/LoadingAnimation.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { motion } from "framer-motion";
+
+const loadingContainer = {
+  width: "5rem",
+  height: "5rem",
+  display: "flex",
```
Comment (mihaelKampic) [inline src/components/LoadingAnimation.js:7]: It's nice to have a loader thats not just a normal circular loader 😁

**Review by MihaelsGit — APPROVED** [review]
Looks good now 👍 
I like the error handling 👏

**Review by IvanJuricic — APPROVED** [review]

**Pull request merged**